### PR TITLE
JS: don't initialize sanitizer-guards in the standard library

### PR DESCRIPTION
**Pull request by erik-krogh**
I did an experiment to [revert the cached stages pattern](https://github.com/github/codeql/pull/8779).  
That gave a whole bunch of QL-for-QL warnings from `ql/abstract-class-import`.  

These warnings are caused by us having sanitizer-guards that were defined in `Xss.qll`.  

I moved things around, including refactoring the old `Xss.qll` where I moved things out into `Query.qll`/`Customizations.qll` files so they match the other queries.  
(Turns out the `js/html-constructed-from-input` query relied on the previous behavior :scream_cat:). 

--- 

The change in results for ATM is caused by: https://github.com/github/codeql/pull/8783/commits/76bf8dec8f8921f57eeec1f770f7ddc728f93916  
The results generally look better (the majority is removal of FPs).    
This PR only affects the `XSS` queries, but the changed ATM results are for `TaintedPath`.   
So the results seem to be a bug caused by importing too many classes.  
So I think we can conclude that this PR is a bug-fix for ATM. 

--- 

[The evaluation looks good](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/jsAbstractBi__nightly-old__code-scanning__1/reports).  
We gained one new result. Which is caused by some barrier-guards no longer falsely limiting the flow.    
Also a slight performance improvement :rocket: 

--- 

I also made some small drive-by fixes of `ql/abstract-class-import` to remove some benign results.

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll [hunk 1]
```diff
@@ -335,4 +328,16 @@ module DomBasedXss {
 
   /** Gets the flow-label representing tainted values where the prefix is attacker controlled. */
   PrefixString prefixLabel() { any() }
+
+  /**
+   * A sanitizer that blocks the `PrefixString` label when the start of the string is being tested as being of a particular prefix.
+   */
+  abstract class PrefixStringSanitizer extends TaintTracking::SanitizerGuardNode,
+    TaintTracking::LabeledSanitizerGuardNode instanceof StringOps::StartsWith {
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll:339]: Is there a need to extend both `SanitizerGuardNode` and `LabeledSanitizerGuardNode`?

File: javascript/ql/lib/semmle/javascript/security/dataflow/StoredXssCustomizations.qll [hunk 2]
```diff
@@ -5,6 +5,11 @@
 
 import javascript
 
+/**
+ * Provides sources, sinks, and sanitizers for reasoning about
+ * cross-site scripting vulnerabilities where the taint-flow passes through a thrown
+ * exception.
+ */
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/security/dataflow/StoredXssCustomizations.qll:12]:
```suggestion
/**
 * Provides default sources, sinks and sanitizers for reasoning about
 * stored cross-site scripting vulnerabilities.
 */
```
Mentions a "thrown exception". I've just suggested the same QLDoc from the file.

File: javascript/ql/lib/change-notes/2022-04-22-xss-library.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+category: deprecated
+---
+* The `ReflectedXss`, `ReflectedXss`, `StoredXss`, `XssThroughDom`, and `ExceptionXss` modules from `Xss.qll` have been deprecated.
```
Comment (asgerf) [inline javascript/ql/lib/change-notes/2022-04-22-xss-library.md:4]:
```suggestion
* The `ReflectedXss`, `StoredXss`, `XssThroughDom`, and `ExceptionXss` modules from `Xss.qll` have been deprecated.  
```

**Review by asgerf — COMMENTED** [review]
Awesome work! 💪

File: javascript/ql/lib/semmle/javascript/security/dataflow/StoredXssCustomizations.qll (hunk 2 above)
Comment (erik-krogh) [reply javascript/ql/lib/semmle/javascript/security/dataflow/StoredXssCustomizations.qll:12]: Right. I accidentally copy-pasted from the wrong place.

File: javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll (hunk 1 above)
Comment (erik-krogh) [reply javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll:339]: Probably not :+1:

Comment (erik-krogh) [reply javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll]: Yep, the tests still pass.

**Review by henrymercer — APPROVED** [review]
ML-powered queries 👍

**Comment by TomBolton** [comment]
@henrymercer how do feel if we merge this PR? I would like to use the merge commit on `main` to start preparing a PR to update the QL SHAs in backend (including the worsening commits)

**Pull request merged**